buildSrc/build.gradle

```diff
@@ -0,0 +1,25 @@
+plugins {
+    id 'groovy-gradle-plugin'
+}
+
+repositories {
+    mavenCentral()
+    maven {
+        url 'https://broadinstitute.jfrog.io/broadinstitute/libs-release-local/'
+    }
+    maven {
+        url 'https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot-local/'
+    }
+    gradlePluginPortal()
+}
+
+dependencies {
+    implementation 'com.diffplug.spotless:spotless-plugin-gradle:6.11.0'
+    implementation 'com.srcclr.gradle:com.srcclr.gradle.gradle.plugin:3.1.12'
+    implementation 'org.sonarqube:org.sonarqube.gradle.plugin:4.2.1.3168'
```

Comment:
Will these be included in, or required by, the final jar we publish to artifactory? I haven't tried it myself to see what the results are. If these ARE included or required, can that be prevented, potentially by choosing a configuration other than `implementation`? We want the published jar to be as small and as dependency-free as possible; and these are only used by build tasks, not at runtime, correct?

I'm fine fixing this in a follow-on PR if you want to defer this work.

Reply:
Great point, I will make sure this is covered with [AJ-1095](https://broadworkbench.atlassian.net/browse/AJ-1095)

Reply:
This file (`buildSrc/build.gradle`) is different than other gradle project configuration files. It's (normally) only used to configure plugins, and its "dependencies" block only affects the plugins loaded by gradle, not compile or runtime dependencies used when running gradle tasks.

So these dependencies have no effect on the jar published to artifactory. They're here so when a plugin is used elsewhere in the project you don't need to specify the version. It provides a way to make sure that the same plugin versions are used in all subprojects.

With that in mind, the `repositories` block should only include those that are needed to load plugins. In my project I use:
```
repositories {
    maven {
        url 'https://broadinstitute.jfrog.io/artifactory/plugins-snapshot'
    }
    gradlePluginPortal()
}
```
The artifactory configuration is needed for the `terra-test-runner` plugin which is locally published. If you aren't using that plugin then `gradlePluginPortal()` should be sufficient.

Also I'd remove the `test` block below. I don't think it has any effect, and it should go in one of the plugin subprojects instead.